### PR TITLE
fix: error handler does not throw errors

### PR DESCRIPTION
**Pull request by akshbhu**
#### Description of changes
Cherry-pick https://github.com/aws-amplify/amplify-cli/commit/f4b9e7bd8ec995a35ac8def0b59e8229efc88729
for sev2 release

#### Description of how you validated changes
e2e: https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/11311/workflows/2fdce27c-50da-44d9-80a2-0ad95a8b159e/jobs/391167
https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/11311/workflows/2fdce27c-50da-44d9-80a2-0ad95a8b159e

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/exception-handler/amplify-exception-handler.ts [hunk 1]
```diff
@@ -41,20 +41,60 @@ export const handleException = async (exception: unknown): Promise<void> => {
...
+ * @param errorMessagePrefix - error message prefix before the thrown error is printed
+ */
+const executeSafely = async (functionToExecute: () => Promise<void> | void, errorMessagePrefix: string): Promise<void> => {
+  try {
+    await functionToExecute();
+  } catch (e) {
+    // Log the error, but do not reject the promise
+    printer.error(`${errorMessagePrefix}: ${e?.message || e}`);
+  }
+};
```
Comment (jhockett) [inline packages/amplify-cli/src/exception-handler/amplify-exception-handler.ts:96]: Something's wrong here, this function shouldn't be duplicated

Comment (akshbhu) [reply packages/amplify-cli/src/exception-handler/amplify-exception-handler.ts:96]: Yup , fixing it

**Comment by sobolk** [comment]
Should we just cherry pick this into hotfix branch https://github.com/aws-amplify/amplify-cli/commit/f4b9e7bd8ec995a35ac8def0b59e8229efc88729 ?

**Review by awsluja — APPROVED** [review]

**Review by edwardfoyle — APPROVED** [review]

**Review by jhockett — APPROVED** [review]

**Review by edwardfoyle — CHANGES_REQUESTED** [review]
The logger bit on line 58 needs to be removed. The logging step was added by a different change on dev and then modified in this commit but the change to add the logger doesn't exist on hotfix.

We'll probably have a merge conflict here when merging back to dev but we should remove the logger step here for now

**Review by edwardfoyle — APPROVED** [review]

**Review by jhockett — APPROVED** [review]

**Pull request merged**